clients/cobol-lsp-vscode-extension/src/__tests__/commands/RunAnalysisCLITest.spec.ts

```diff
@@ -0,0 +1,283 @@
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Broadcom, Inc. - initial API and implementation
+ */
+
+import { RunAnalysis } from "../../commands/RunAnalysisCLI";
+import * as vscode from "vscode";
+
+jest.mock("vscode", () => ({
+  commands: {
+    registerCommand: jest
+      .fn()
+      .mockImplementation((command, callback) => callback()),
+    executeCommand: jest.fn(),
+  },
+  extensions: {
+    getExtension: jest.fn().mockReturnValue({ extensionPath: "/test" }),
+  },
+  FileType: {
+    Directory: 2,
+  },
+  languages: {
+    registerCodeActionsProvider: jest.fn(),
+    registerCompletionItemProvider: jest.fn(),
+  },
+  window: {
+    activeTextEditor: {
+      document: {
+        uri: {
+          path: "/storagePath",
+          fsPath: "/storagePath",
+          scheme: "file",
+        },
+        getText: jest.fn(),
+      },
+    },
+    setStatusBarMessage: jest
+      .fn()
+      .mockImplementation(
+        async (_text: string, _hideWhenDone: Thenable<any>) => {
+          return Promise.resolve(true);
+        },
+      ),
+    showErrorMessage: jest.fn().mockReturnValue("test"),
+    showInformationMessage: jest
+      .fn()
+      .mockImplementation((message: string) => Promise.resolve(message)),
+    showQuickPick: jest.fn(),
+    onDidChangeActiveTextEditor: jest.fn(),
+    createQuickPick: jest
+      .fn()
+      .mockReturnValue({ onDidChangeSelection: jest.fn(), show: jest.fn() }),
+    createOutputChannel: jest.fn().mockReturnValue({
+      appendLine: jest.fn(),
+    }),
+    terminals: {
+      find: jest.fn(),
+    },
+    createTerminal: jest.fn().mockReturnValue({
+      sendText: jest.fn(),
+      show: jest.fn(),
+    }),
+  },
+  workspace: {
+    getConfiguration: jest.fn().mockReturnValue({
+      get: jest.fn().mockReturnValue(9),
+    }),
+    getWorkspaceFolder: jest.fn().mockReturnValue({ name: "workspaceFolder1" }),
+    onDidChangeConfiguration: jest
+      .fn()
+      .mockReturnValue("onDidChangeConfiguration"),
+    workspaceFolders: [{ uri: { fsPath: "ws-path" } } as any],
+    fs: {
+      createDirectory: jest.fn(),
+      readDirectory: jest.fn().mockReturnValue([]),
+      stat: jest.fn().mockReturnValue(2),
+      writeFile: jest.fn(),
+    },
+  },
+  Uri: {
+    file: jest.fn().mockReturnValue("workspaceFolder2"),
+    parse: jest.fn().mockReturnValue({ path: "/storagePath" }),
+    joinPath: jest.fn().mockImplementation((inputUri: vscode.Uri, path) => {
+      return {
+        path: inputUri.path + path,
+        fsPath: inputUri.fsPath + path,
+      };
+    }),
+  },
+}));
+
+const context: any = {
+  extensionUri: { fsPath: "/test" },
+  globalStorageUri: { fsPath: "/storagePath" },
+  subscriptions: [],
+};
+
+describe("Test Analysis CLI command functionality", () => {
+  test("Cobol Analysis - Java", async () => {
+    const testAnalysis = new RunAnalysis(
+      vscode.Uri.parse(context.globalStorageUri),
+      vscode.Uri.parse(context.extensionUri),
```

Comment:
```suggestion
      context.globalStorageUri,
      context.extensionUri,
```
the same change should be done in all the other test cases as well.